lib/galaxy/datatypes/binary.py

```diff
@@ -169,6 +171,88 @@ def _get_samtools_version( self ):
+        input_file_bai = ""
+        for key, value in ds.metadata.items():
+            if isinstance( value, galaxy.model.MetadataFile ):
+                input_file_bai = value.file_name
+
+        # Link files with bam extensions in order to fulfil some tools requirements
+        commands = ["ln -sf " + input_file + " " + input_file + ".bam",
+                    "ln -sf " + input_file_bai + " " + input_file + ".bam.bai"]
+        for cmd in commands:
+            if 0 != os.system(cmd):
+                raise Exception("Executing '%s' failed" % cmd)
+
+        if 'split_mode' not in split_params:
+            raise Exception('Tool does not define a split mode')
+        elif split_params['split_mode'] == 'by_rname':
+            log.debug("Attempting to split BAM file %s by chromosome", input_file)
+
+            # First get bam header
+            params = ["samtools", "view", "-H", input_file]
+            output = subprocess.Popen( params, stderr=subprocess.PIPE, stdout=subprocess.PIPE ).communicate()[0]
+            output = output.split("\n")
+            chr_list = []
+            for line in output:
+                fields = line.strip().split("\t")
+                if fields[0].startswith("@SQ") and fields[1].startswith("SN:"):
+                    chr_list.append(fields[1].split("SN:")[1])
+
+            for chrName in chr_list:
+                try:
+                    part_dir = subdir_generator_function()
+                    base_name = os.path.basename(input_file)
+                    part_path = os.path.join(part_dir, base_name)
+
+                    split_data = dict(class_name='%s.%s' % (cls.__module__, cls.__name__),
+                                      output_name=part_path,
+                                      input_name=input_file,
+                                      args=dict(chromosome=chrName))
+                    f = open(os.path.join(part_dir, 'split_info_%s.json' % base_name), 'w')
+                    json.dump(split_data, f)
+                    f.close()
+
+                except Exception, e:
+                    log.error("Error: " + str(e))
+                    raise
+        else:
+            raise Exception('Unsupported split mode %s' % split_params['split_mode'])
+    split = classmethod(split)
+
+    def process_split_file(data):
+
+        args = data['args']
+        input_name = data['input_name']
+        output_name = data['output_name']
+        chromosome = args['chromosome']
+
+        commands = Bam.get_split_commands_chromosome(input_name, output_name, chromosome)
+        for cmd in commands:
+            if 0 != os.system(cmd):
+                raise Exception("Executing '%s' failed" % cmd)
+        return True
+    process_split_file = staticmethod(process_split_file)
+
+    def get_split_commands_chromosome(input_name, output_name, chromosome):
+        commands = [
+            "samtools view -bh " + input_name + ".bam " + chromosome + " > " + output_name
```

Comment:
Again this should be more careful about quoting - probably best to use subprocess.Popen like above with individual arguments in a list (i.e. ``shell=False``). It looks like samtools view has a ``-o`` option - is that viable to use instead of ``>``?